HopsanGUI/HcomHandler.cpp

```diff
@@ -6439,154 +6442,126 @@ void HcomHandler::evaluateExpression(QString expr, VariableType desiredType)
                 windowingFunction = RectangularWindow;
+                ++i;
             }
-            else if(windowingFuncArg.toLower() == "flattop") {
+            else if(splitArgs[i].toLower() == "flattop") {
                 windowingFunction = FlatTopWindow;
-            }
-            else {
-                HCOMERR("Unknown windowing function: "+windowingFuncArg);
-                mAnsType = Undefined;
-                return;
+                ++i;
             }
         }
 
-        //Parse window min and max time arguments
-        double minTime, maxTime;
-        if(!minTimeArg.isEmpty()) {
-            if(maxTimeArg.isEmpty()) {
-                HCOMERR("Maximum time for windowing is required when minimum time is specified");
-                mAnsType = Undefined;
-                return;
-            }
+        double minTime=-std::numeric_limits<double>::max();
+        double maxTime=std::numeric_limits<double>::max();
+
+        if(splitArgs.size() > i+1) {
             bool ok;
-            minTime = getNumber(minTimeArg, &ok);
+            minTime = getNumber(splitArgs[i], &ok);
             if(!ok) {
-                HCOMERR("Unknown time limit: "+minTimeArg);
+                HCOMERR("Unknown minimum time limit: "+splitArgs[i]);
                 mAnsType = Undefined;
                 return;
             }
-            maxTime = getNumber(maxTimeArg, &ok);
+            maxTime = getNumber(splitArgs[i+1], &ok);
             if(!ok) {
-                HCOMERR("Unknown time limit: "+maxTimeArg);
+                HCOMERR("Unknown maximum time limit: "+splitArgs[i+1]);
                 mAnsType = Undefined;
                 return;
             }
+            i+=2;
         }
 
-        if(typeArg.isEmpty()) {
-            mAnsVector = pDataVar->toFrequencySpectrum(pTimeVar, PowerSpectrum);
+        if(splitArgs.size() > i) {
+            HCOMERR("Unknown argument: "+splitArgs[i]);
+            mAnsType = Undefined;
+            return;
         }
-        else if(windowingFuncArg.isEmpty()) {
-            mAnsVector = pDataVar->toFrequencySpectrum(pTimeVar, type);
+
+        mAnsType = DataVector;
+        mAnsVector = pVar->toFrequencySpectrum(pTimeVar, type, windowingFunction, minTime, maxTime);
+        return;
+    }
+    else if(isHcomFunctionCall("rms", expr))
+    {
+        QString args = expr.mid(4, expr.size()-5);
+        QStringList splitArgs = SymHop::Expression::splitWithRespectToParentheses(args, ',');
+        if(splitArgs.size() != 1) {
+            HCOMERR("Wrong number of arguments provided for rms function.\n"+mLocalFunctionDescriptions.find("fft").value().second);
+            mAnsType = Undefined;
+            return;
         }
-        else if(minTimeArg.isEmpty()) {
-            mAnsVector = pDataVar->toFrequencySpectrum(pTimeVar, type, windowingFunction);
+        mAnsType = Scalar;
+        const QString varName = args.section(",",0,0).trimmed();
```

Comment:
rms seems to only have one argument, and did we not already split the args above ?